File: app/src/main/java/ua/ck/zabochen/englishverbs/dagger/AppComponent.java
package ua.ck.zabochen.englishverbs.dagger;

import javax.inject.Singleton;

import dagger.Component;
import ua.ck.zabochen.englishverbs.view.main.MainPresenter;
import ua.ck.zabochen.englishverbs.view.verbfull.VerbFullPresenter;
import ua.ck.zabochen.englishverbs.view.verblist.VerbListPresenter;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(MainPresenter mainPresenter);
    void inject(VerbListPresenter verbListPresenter);
    void inject(VerbFullPresenter verbFullPresenter);

}

File: app/src/main/java/ua/ck/zabochen/englishverbs/dagger/component/AppComponent.kt
package ua.ck.zabochen.englishverbs.dagger.component

import android.content.Context
import dagger.Component
import ua.ck.zabochen.englishverbs.dagger.module.ApplicationContextModule
import javax.inject.Singleton

@Component(modules = [ApplicationContextModule::class])
@Singleton
interface AppComponent {

    // Dependent components
    fun activityComponentBuilder(): ActivityComponent.Builder

    // Available dependencies for child components
    fun applicationContext(): Context

}